### PR TITLE
[collections] Check for Invalid Push Size of Concurrent Ring Buffer

### PR DESCRIPTION
**Pull request by ppenna**
## Description

This PR fixes invalid push size for concurrent ringle buffers. It also adds more test units to this data structure.

### CODE REVIEW

File: src/rust/collections/concurrent_ring.rs [hunk 1]
```diff
@@ -506,6 +500,58 @@ mod test {
         Ok(ring)
     }
 
+    /// Pushes data to a ring buffer until it is full.
+    fn make_full_ring_buffer() -> Result<ConcurrentRingBuffer> {
```
Comment (iyzhang) [inline src/rust/collections/concurrent_ring.rs:504]: could call this `fill_ring_buffer`

File: src/rust/collections/concurrent_ring.rs [hunk 2]
```diff
@@ -506,6 +500,58 @@ mod test {
...
+        let mut npushes: usize = 0;
+        let byte: [u8; 1] = [0xff];
+        let ring: ConcurrentRingBuffer = do_new()?;
+        while ring.remaining_capacity() > HEADER_SIZE {
+            do_sucess_push_bytes(&ring, &byte)?;
```
Comment (iyzhang) [inline src/rust/collections/concurrent_ring.rs:509]: Success has 2 Cs

**Review by iyzhang — APPROVED** [review]

File: src/rust/collections/concurrent_ring.rs (hunk 1 above)
Comment (ppenna) [reply src/rust/collections/concurrent_ring.rs:504]: Sure.

File: src/rust/collections/concurrent_ring.rs (hunk 2 above)
Comment (ppenna) [reply src/rust/collections/concurrent_ring.rs:509]: Thanks.

**Pull request merged**